### PR TITLE
Fix for the redirect follow of CA request

### PR DESCRIPTION
**Pull request by heidijinxujia**
1. fix the browser launching 
2. add the logic to launch the company portal if it is already installed

Finished the e2e testing w/ and w/o company portal installed to do the MDM auth flow.

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java [hunk 1]
```diff
@@ -206,18 +208,44 @@ protected boolean processRedirectUrl(@NonNull final WebView view, @NonNull final
...
+        }
+
         view.stopLoading();
         Intent resultIntent = new Intent();
         getCompletionCallback().onChallengeResponseReceived(AuthorizationStrategy.UIResponse.AUTH_CODE_CANCEL, resultIntent);
         return true;
     }
 
+    private void openLinkInBrowser(final String url) {
+        //Open url link in browser
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java]: Can we add a log statement here

File: common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java [hunk 2]
```diff
@@ -206,18 +208,44 @@ protected boolean processRedirectUrl(@NonNull final WebView view, @NonNull final
...
-                .replace(AuthenticationConstants.Broker.BROWSER_EXT_PREFIX, "https://");
-        final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(link));
-        getActivity().startActivity(intent);
+        final PackageHelper packageHelper = new PackageHelper(getActivity().getApplicationContext());
+        if (url.startsWith(AuthenticationConstants.Broker.BROWSER_DEVICE_CA_URL)
+                && packageHelper.isPackageInstalledAndEnabled(AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME)) {
+            Logger.verbose(TAG, "It is a device CA request. Company Portal is installed.");
+            try {
+                Logger.verbose(TAG, "Sending intent to launch the CompanyPortal.");
+                Intent intent = new Intent();
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java]: nit: final

File: common/src/main/java/com/microsoft/identity/common/adal/internal/util/PackageHelper.java [hunk 3]
```diff
@@ -151,6 +151,27 @@ public int getUIDForPackage(final String packageName) {
...
     }
 
+    /**
+     * Check if the given package is installed and enabled.
+     *
+     * @param packageName the package name to look up.
+     * @return true if the package is installed and enabled. Otherwise, returns false.
+     */
+    public Boolean isPackageInstalledAndEnabled(final String packageName) {
+        String methodName = "isPackageInstalledAndEnabled";
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/util/PackageHelper.java]: nit: final for variables in this method

File: common/src/main/java/com/microsoft/identity/common/adal/internal/util/PackageHelper.java [hunk 4]
```diff
@@ -151,6 +151,27 @@ public int getUIDForPackage(final String packageName) {
...
+        Boolean enabled = false;
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/util/PackageHelper.java]: Can we use primitive `boolean` here?

**Comment by heidijinxujia** [comment]
@kreedula Thank you! All comments addressed.

**Pull request merged**